### PR TITLE
First app structure

### PR DESCRIPTION
**Pull request by donmikelee**
Master, this is my idea for the arrangement of structure elements and controls with functions.

The elements folder is a simple HTML structure without complicated functions.

Controls contain HTML elements with advanced functions and states.

Please pay attention to how I wrote the class interface for the Header component. Is this correct and practical?

Do you have any ideas on what tools I could use to make the project impressive and maintain the level of an advanced junior? :D

### CODE REVIEW

File: Countries-searcher/src/App.tsx [hunk 1]
```diff
@@ -1,33 +1,13 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
```
Comment (AdamKniec) [inline Countries-searcher/src/App.tsx:3]: I think that You can remove this logo, right?

File: Countries-searcher/src/components/controls/InputSearch/InputSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const InputSearch = () => {
+    return ( 
+        <>
+            <span>IconSearch</span>
+            <input type="text" placeholder="Search for a country..."/>
```
Comment (AdamKniec) [inline Countries-searcher/src/components/controls/InputSearch/InputSearch.tsx:5]: I think that you forgot about the label and accessibility here

**Comment by AdamKniec** [comment]
image
I know that this was an initial commit but I personally think that it'd be better to leave the page blank for now instead of committing totally unfinished components. Hint for the future -> try to finish some component / functionality and only after that create the PR

File: Countries-searcher/src/components/controls/InputSearch/InputSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const InputSearch = () => {
+    return ( 
+        <>
+            <span>IconSearch</span>
```
Comment (AdamKniec) [inline Countries-searcher/src/components/controls/InputSearch/InputSearch.tsx:4]: I saw nothing like that on the design

File: Countries-searcher/src/components/elements/Content/Content.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import DropdownRegions from "../../controls/DropdownRegions/DropdownRegions";
+import InputSearch from "../../controls/InputSearch/InputSearch";
+import CountryCard from "../CountryCard/CountryCard";
+
+const Content = () => {
+    return (  
+        <div>
+            Main Content
```
Comment (AdamKniec) [inline Countries-searcher/src/components/elements/Content/Content.tsx:8]: I saw nothing like that on the design

File: Countries-searcher/src/components/elements/Content/Content.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import DropdownRegions from "../../controls/DropdownRegions/DropdownRegions";
+import InputSearch from "../../controls/InputSearch/InputSearch";
+import CountryCard from "../CountryCard/CountryCard";
+
+const Content = () => {
```
Comment (AdamKniec) [inline Countries-searcher/src/components/elements/Content/Content.tsx:5]: Content ? What content :D please come up with more descriptive name

File: Countries-searcher/src/components/elements/Header/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+const cssClasses: HeaderClasses = {
+    mainClass: "csHeader",
+};
+
+const Header = () => {
+
+    const { mainClass } = cssClasses
+    return ( 
+        <div className={mainClass}>
```
Comment (AdamKniec) [inline Countries-searcher/src/components/elements/Header/Header.tsx:12]: Why did you use div here? Maybe there is a better alternative?

File: Countries-searcher/src/components/elements/Header/Interface.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+interface HeaderClasses {
```
Comment (AdamKniec) [inline Countries-searcher/src/components/elements/Header/Interface.ts:1]: I see no point in creating a separate file with 3 lines interface for now. Move it to the component file for now

File: Countries-searcher/src/components/elements/Header/Header.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import ThemeSwitcher from '../../controls/ThemeSwticher/ThemeSwitcher.tsx';
+import './Interface.ts'
+
+const cssClasses: HeaderClasses = {
+    mainClass: "csHeader",
```
Comment (AdamKniec) [inline Countries-searcher/src/components/elements/Header/Header.tsx:5]: What does cs stand for?

File: Countries-searcher/src/components/elements/Header/Header.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import ThemeSwitcher from '../../controls/ThemeSwticher/ThemeSwitcher.tsx';
+import './Interface.ts'
+
+const cssClasses: HeaderClasses = {
```
Comment (AdamKniec) [inline Countries-searcher/src/components/elements/Header/Header.tsx:4]: What;s the idea behind this object?

File: Countries-searcher/src/components/elements/CountryCard/CountryCard.tsx [hunk 10]
```diff
@@ -0,0 +1,7 @@
+const CountryCard = () => {
```
Comment (AdamKniec) [inline Countries-searcher/src/components/elements/CountryCard/CountryCard.tsx:1]: Empty and unfinished : (

File: Countries-searcher/src/components/controls/ThemeSwticher/ThemeSwitcher.tsx [hunk 11]
```diff
@@ -0,0 +1,7 @@
+const ThemeSwitcher = () => {
```
Comment (AdamKniec) [inline Countries-searcher/src/components/controls/ThemeSwticher/ThemeSwitcher.tsx:1]: Empty and unfinished : (

File: Countries-searcher/src/components/controls/DropdownRegions/DropdownRegions.tsx [hunk 12]
```diff
@@ -0,0 +1,7 @@
+const DropdownRegions = () => {
+    return ( 
+        <select name="regions"></select>
```
Comment (AdamKniec) [inline Countries-searcher/src/components/controls/DropdownRegions/DropdownRegions.tsx:3]: Empty select?

File: Countries-searcher/src/components/elements/Header/Header.tsx (hunk 8 above)
Comment (donmikelee) [reply Countries-searcher/src/components/elements/Header/Header.tsx]: cs is a shortcut of countries-searcher

**Pull request closed without merging**